### PR TITLE
fix: move webpack to a peer dependency and allow webpack 2

### LINKED ISSUE #38: Webpack 2 support coming soon?
At least for some beta channel maybe?

### PR DESCRIPTION
**Pull request by mattlewis92**
Using this plugin with webpack 2 currently installs 2 versions of webpack.

Resolves #38.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,14 +23,14 @@
   },
   "homepage": "https://github.com/vieron/stylelint-webpack-plugin#readme",
   "peerDependencies": {
-    "stylelint": "^7.0.1"
+    "stylelint": "^7.0.1",
+    "webpack": "^2 || ^2.2.0-rc.0 || ^2.1.0-beta.0 || ^1.13.2"
```
Comment (JaKXz) [inline package.json]: I'd rather just have `^2.0.0 || ^1.13.2` here... I would think it's better to just "officially" support stable versions. With `npm` >=3 or `yarn` you won't get peerDeps installed anyway so you can still choose what version you want.

Comment (mattlewis92) [inline package.json]: There are no breaking changes between webpack 1 and webpack 2's API that would affect this plugin, it's very safe to support the beta versions of webpack, and gets rid of the annoying peer dependency warning on install.

**Comment by mattlewis92** [comment]
@JaKXz I've added a simple npm script to test against both v1 and v2 of webpack, LMK what you think.

File: package.json (hunk 1 above)
Comment (JaKXz) [inline package.json]: I see, because the `rc` and `beta` versions do not fall under semver technically so they aren't resolved by `^2.0.0`? Fair enough.

**Comment by mattlewis92** [comment]
Yup that's correct, this is how all the webpack plugins have their peer dependency set under the webpack org

**Comment by mattlewis92** [comment]
@JaKXz I just spotted there was a merge conflict with master which I've now resolved, would you like me to make any more changes before this can be merged? 😃

**Comment by JaKXz** [comment]
@mattlewis92 thanks for fixing that!

I'm still not sure about the way to test the different versions of webpack though. Maybe this should be a bigger discussion with @sokra and other plugin maintainers who want to maintain backwards compatibility?

**Comment by mattlewis92** [comment]
Given that webpack 2 is now in RC and there are no more breaking changes planned (and previously none that have affected this plugin) - I wouldn't worry too much about it. AFAIK the plugin / loader API will always be pretty stable to avoid breaking the webpack ecosystem - the majority of breaking changes are in the EU's webpack config itself. Maybe @TheLarkInn can give some more insight into this though?

File: package.json [hunk 2]
```diff
@@ -43,11 +43,14 @@
...
     "nyc": "^8.3.1",
-    "xo": "^0.16.0"
+    "xo": "^0.16.0",
+    "webpack": "^1.13.2"
   },
   "scripts": {
     "pretest": "xo",
-    "test": "nyc mocha",
+    "test:webpack1": "npm i webpack@1 && nyc mocha",
+    "test:webpack2": "npm i webpack@beta && nyc mocha && npm i webpack@1",
```
Comment (JaKXz) [inline package.json]: If you don't mind trying these out with `yarn` I think that would make it a bit faster and so on.

**Comment by JaKXz** [comment]
@mattlewis92 thank you for your patience on this! 

I had a look at the _long_ thread that is https://github.com/npm/npm/issues/5499 and found: https://github.com/scott113341/npm-install-version

I think a programmatic solution with `npm-install-version` is the way to go - at first glance it looks like a sustainable way to test against both versions of webpack until webpack 1 support is dropped, which of course isn't happening any time soon.

I really appreciate your work on this - let me know if you want to give that a shot or I can take it on if you're spent. :)

**Comment by mattlewis92** [comment]
@JaKXz I've done a basic implementation using npm-install-version - let me know if that's OK for you

**Comment by JaKXz** [comment]
`+ stylelint-webpack-plugin@0.5.0` published! thanks again!

**Comment by mattlewis92** [comment]
Awesome, thanks! 😃

**Pull request merged**